src/cmakeTaskProvider.ts

```diff
@@ -39,23 +66,30 @@ export class CMakeTaskProvider implements vscode.TaskProvider {
         }
     }
 
-    public updateDefaultTargets(defaultTargets: string[] | undefined) {
+    public updateDefaultTargets(defaultTargets?: string[]) {
         this.defaultTargets = (defaultTargets && defaultTargets.length > 0) ? defaultTargets :
             this.cmakeDriver ? [this.cmakeDriver.allTargetName] : [CMakeTaskProvider.allTargetName];
     }
 
     public async provideTasks(): Promise<CMakeTask[]> {
         // Create a CMake build task
         const result: CMakeTask[] = [];
-        const taskName: string = "CMake: build";
+        this.updateDefaultTargets();
+        // Provide build task.
+        const taskName: string = localizeCommandType(CommandType.build);
```

Comment:
Conisder renaming `taskName` to `commandType` since it is not the entire task name (e.g. label)